internal/controller/nodeagent.go

```diff
@@ -290,7 +290,9 @@ func (r *DataProtectionApplicationReconciler) customizeNodeAgentDaemonset(ds *ap
 	// Update with any pod config values
 	if dpa.Spec.Configuration.NodeAgent.PodConfig != nil {
 		ds.Spec.Template.Spec.Tolerations = dpa.Spec.Configuration.NodeAgent.PodConfig.Tolerations
-		ds.Spec.Template.Spec.NodeSelector = dpa.Spec.Configuration.NodeAgent.PodConfig.NodeSelector
+		if len(dpa.Spec.Configuration.NodeAgent.PodConfig.NodeSelector) != 0 {
```

Comment:
why change here?

Reply:
alright, saw that now velero deployment and node-agent daemonSet have nodeSelector `"kubernetes.io/os": "linux"`

but shouldn't we change logic here then? instead of changing nodeSelector, append to it, to do not overwrite what velero set?

Reply:
@mateusoliveira43 That's a good question. Maybe we should append here instead? The intent here is to avoid windows nodes. Is that a concern here? @shubham-pampattiwar @kaovilai thoughts?

Reply:
ok how about we drop the windows related selectors (if they exist) and keep everything else and then append to it ?

Reply:
@shubham-pampattiwar Currently I'm thinking that we just overwrite -- that way we have a workaround if the velero-generated selectors don't work in a particular customer environment. If the user really needs the velero selector components on top of their own, they can include it in their DPA.

Reply:
yeah that works too, DPA to the rescue